api/permissions.py

```diff
@@ -51,7 +51,10 @@ def has_object_permission(self, request, view, obj):
 class GradeBookPermission(permissions.IsAuthenticated):
     def has_object_permission(self, request, view, obj):
         if request.method == "GET":
-            return obj.has_edit_permission(request.user)
+            if request.user.is_student:
```

Comment:
I get this but it scares me in a future way - like this permission implies that just being a student that is registered should grant access to APIs with this permission

I know that in the endpoint itself, we only show students their own data, but if someone tried to reuse this permission based on its name, we have have some issues

Overall, maybe it's fine but definitely toss a # TODO comment on this and/or a ticket

(Bigger solution I would suggest is to have separate endpoints for the "see all gradebook data" vs "see MY gradebook data" and have different permissions for each, but in the interest of time, don't sweat this too too much)

Reply:
You need to disallow students from the list API. The list API doesn't consider the obj permissions